test/tools/unified-spec-runner/match.ts

```diff
@@ -225,9 +249,7 @@ export function specialCheck(
     // $$unsetOrMatches
     if (actual === null || actual === undefined) return;
     else {
```

Comment:
Depth represents how many levels deep in an object we are.

For `$$unsetOrMatches`, we don't want to update the depth because the `$$unsetOrMatches` operator isn't part of the document we're evaluating - it's more of a "meta" key telling us how to evaluate the object.

Reply:
Comment no longer relevant now that I've refactored to use `isRootDocument` instead of `depth`.